From f28b6a3339c14c9deb4b44a5fe1c583a20b095fa Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 28 Aug 2014 03:04:23 -0400
Subject: [acpi] Clean up error code path in acpi_unbind_one()

Message-id: <20140828030423.2117.97073.stgit@gir.stowe>
Patchwork-id: 89296
O-Subject: [RHEL7.1 PATCH 08/30] ACPI: Clean up error code path in acpi_unbind_one()
Bugzilla: 1128632
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1128632
Upstream Status: 38e88839eff8a3d2e8d3bcc2ad833fe51cca0496

commit 38e88839eff8a3d2e8d3bcc2ad833fe51cca0496
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Wed Aug 7 01:22:51 2013 +0200

    ACPI: Clean up error code path in acpi_unbind_one()

    The error code path in acpi_unbind_one() is unnecessarily complicated
    (in particular, the err label is not really necessary) and the error
    message printed by it is inaccurate (there's nothing called
    'acpi_handle' in that function), so clean up those things.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Acked-by: Toshi Kani <toshi.kani@hp.com>
    Acked-by: Yasuaki Ishimatsu <isimatu.yasuaki@jp.fujitsu.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/glue.c b/drivers/acpi/glue.c
index 570628e..dcba319 100644
--- a/drivers/acpi/glue.c
+++ b/drivers/acpi/glue.c
@@ -284,8 +284,10 @@ int acpi_unbind_one(struct device *dev)
   return 0;
 
  status = acpi_bus_get_device(ACPI_HANDLE(dev), &acpi_dev);
- if (ACPI_FAILURE(status))
-  goto err;
+ if (ACPI_FAILURE(status)) {
+  dev_err(dev, "Oops, ACPI handle corrupt in %s()\n", __func__);
+  return -EINVAL;
+ }
 
  mutex_lock(&acpi_dev->physical_node_lock);
 
@@ -307,12 +309,7 @@ int acpi_unbind_one(struct device *dev)
   }
 
  mutex_unlock(&acpi_dev->physical_node_lock);
-
  return 0;
-
-err:
- dev_err(dev, "Oops, 'acpi_handle' corrupt\n");
- return -EINVAL;
 }
 EXPORT_SYMBOL_GPL(acpi_unbind_one);
 
-- 
1.7.1